### DISCUSSION TITLE
Unable to clear Error code ESO2

### DISCUSSION THREAD

**Discussion by Geo-Ron** [jazzmonger/wood-pellet-stove-with-TYWE1S-Tuya-chip#39] [Q&A] [answered]
Hi Jeff,
I am really sorry to bother you with this, but I am running out of options for now...

The stove was not working well. Pressing the thumbs up showed a room temperature of 84 degrees celsius.
I noticed someone cut and resoldered the room temp sensor.
After investigation I found out the sensor was a 10K NTC. Comparing values it should be a 100K NTC.
I mounted a 100K resistor instead and it started working OK and showing the correct temperature of 25 degrees Celsius.
While working on the sensor it showed error ESO3, which is a correct error.

But then the stove raised an error ESO2 and I can not get rid of it. Nor pressing the thumbs-up button, cutting the power or **exchanging the 3 temp sensors**. It always comes up with the ESO2 error. Measuring the resistance of all the sensors show ok values in relation to a 100K NTC sensor.

Got any tips for me?

I removed the main MCU and measured the traces to the ICU and they all seem ok to me. No visual clues on the board what could be wrong.

![image](https://github.com/user-attachments/assets/48f88638-86e2-4a2d-8bd1-80c5e8f19fbd)

**Comment by jazzmonger** [comment]
ESO2 is an open error.  I don't recall which temp sensor, but there are only 3. On the bench,  I took a 250k pot if memory serves and connected it to the wires for the sensor and adjusted until the error went away.

EDIT: it's the "protection" sensor on the fire box 

<img width="702" alt="image" src="https://github.com/user-attachments/assets/d6ef2c46-3102-4192-b2b7-f3c0f6ace094">
There are 3 temp sensors:
- room sensor
- Exhaust sensor
- Firebox sensor

as far as I could ever tell, they were all the same sensor meaning I could interchange them at will on the bench.

**Comment by Geo-Ron** [comment]
I read your docs.

It looks like all the (temp) sensors do not give a reading.

Room temp (-20 was showing when the room sensor was disconnected, together with ESO3):
![image](https://github.com/user-attachments/assets/7215544a-c5a2-496c-a39c-d89448acd508)

Other
![image](https://github.com/user-attachments/assets/6efed93f-f598-4338-83f9-9f3950ea9601)
![image](https://github.com/user-attachments/assets/244ac350-e354-4307-9565-d961873622a8)

I checked the connections on the MCU, but was unable to find something faulty
![image](https://github.com/user-attachments/assets/58782e94-dc56-4247-b627-dc4bd5cd00d6)

**Comment by jazzmonger** [comment]
Sounds like it very well could be a problem on the main MCU board.  If you've continuity checked the connections all the way to the pin on the MCU, there's something else wrong. Possibly a bad MCU... 
If you were in the USA I'd tell you to call cleaveland ironworks and get a replacement board.  I've done it a couple times even though I was out of warranty. That's how I got the extra boards to do all my development work. Perhaps there a distributor where you are who you could check with?

**Reply by Geo-Ron** [reply]
Then I will get a replacement MCU board. Too bad they are costly...

**Comment by Geo-Ron** [comment]
Received the 'new' main MCU.
Errors are gone now, but the smoke fan does not start in the cleaning phase. It only hums then errors E5, which is logical.
On shutdown the smoke fan does run, but slower then I am used to.

I am not pleased at this moment...

Pictures of the new board: https://github.com/Geo-Ron/esphome-PelletStove/blob/main/main-control-unit.md#after-replacement-of-the-mcu

**Reply by jazzmonger** [reply]
Its an entirely different board.  2 of the Triacs are missing.  So it very likely has different firmware which might explain why the fan doesn't respond correctly - it was built for an OEM w/ different parts/motors than yours.  I'd contact the company you got it from and see if they can swap it for the correct version.

Where are you located?

**Reply by Geo-Ron** [reply, marked as answer]
I am located at the Netherlands.

I solved the issue by also swapping the display unit.

It has exactly the same hardware, but a different firmware version. The old display has version v02.02 and the newer display has version v3.01.

For now, my problems are solved and I can continue working in the ESP board...

**Comment by jazzmonger** [comment]
Good news!  It’s always something!
